### PR TITLE
Change `None` to `"local_only"` for connection method and expand connection tests

### PR DESCRIPTION
**Pull request by JoeZiminski**
`connection_method` should be set to `None` to use local-only mode #457. However, all all other methods are `str` (e.g. `"gdrive"`). This meant that `get_connection_methods_list()` did not return `None` option leading to asserts being triggered in some cases when in local_only mode (e.g. [here](https://github.com/neuroinformatics-unit/datashuttle/blob/04233d5e67a6c982794ac9c2e3dc15f8227bcca5/datashuttle/tui/tabs/create_folders.py#L336)).

In general it was not good to have local only mode indicated by `None` and led to a lot of workarounds and is confusing for code readers. Now, internally `local_only` is used as the `connection_method` for local only. `None` is still accepted as an alias for `"local_only"` in the make / update config file, and old `None` in config files are now updated on load.

It also centralises and extends the use of a custom type for the connection methods.

As well as this refactoring, tests are added to 1) check the alias in the config update methods 2) add a backward compatibility test to ensure `None` is converted to local only in loaded configs. 

Documentation changes are not necessary as the default argument for config making is `local_only` so for local only mode, the argument is not filled it.

### CODE REVIEW

File: tests/tests_regression/test_backwards_compatibility.py [hunk 1]
```diff
@@ -101,13 +101,26 @@ def test_new_connection_methods_for_backwards_compatibility(
             assert key in reloaded_ver_configs
             assert reloaded_ver_configs[key] is None
 
-    def load_and_check_old_version_yamls(
-        self, project, tmp_path, datashuttle_version
+    def test_connection_method_none(self, project, tmp_path):
+        """Check an old version of the configs where `connection_method` can equal `None` is loaded correctly."""
+        reloaded_ver_configs, _ = self.load_old_version_yamls(
+            project, tmp_path, "connection_method_none", check_correct=False
+        )
```
[the remark this answers is not in the document]
Comment (JoeZiminski) [reply tests/tests_regression/test_backwards_compatibility.py]: thanks @cs7-shrey, in the end I refactored this as it was a bit lazy of me, really there should be one function for loading and one for checking. Its a bit more verbose now but definitely clearer. Thanks!
